hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/PrefixParser.java

```diff
@@ -0,0 +1,213 @@
+
+  @Spec
+  private CommandSpec spec;
+
+  @CommandLine.Option(names = {"--db"},
+      required = true,
+      description = "Database File Path")
+  private String dbPath;
+
+  @CommandLine.Option(names = {"--path"},
+      required = true,
+      description = "prefixFile Path")
+  private String filePath;
+
+  @CommandLine.Option(names = {"--bucket"},
+      required = true,
+      description = "bucket in /vol/bucket format")
+  private String bucket;
+
+  @CommandLine.Option(names = {"--volume"},
+      required = true,
+      description = "volume in /vol/bucket format")
+  private String volume;
+
+  public enum Type {
+    BUCKET,
+    INTERMEDIATE_DIR,
+    TERMINAL_FILE,
+    TERMINAL_DIR
+  }
+
+  private long[] numOps;
+
+  public PrefixParser() {
+    this.numOps = new long[Type.values().length];
+  }
+
+  public String getDbPath() {
+    return dbPath;
+  }
+
+  public void setDbPath(String dbPath) {
+    this.dbPath = dbPath;
+  }
+
+  public void setVolume(String volume) {
+    this.volume = volume;
+  }
+
+  public void setBucket(String bucket) {
+    this.bucket = bucket;
+  }
+
+  public void setFilePath(String filePath) {
+    this.filePath = filePath;
+  }
+
+  @Override
+  public Class<?> getParentType() {
+    return OzoneDebug.class;
+  }
+
+  @Override
+  public Void call() throws Exception {
+    parse();
+    return null;
+  }
+
+  public static void main(String[] args) throws Exception {
+    new PrefixParser().call();
+  }
+
+  public void parse() throws Exception {
+    System.out.println("FilePath is:" + filePath);
+    System.out.println("Db Path is:" + dbPath);
+
+    OzoneConfiguration conf = new OzoneConfiguration();
+    conf.set(OMConfigKeys.OZONE_OM_DB_DIRS, dbPath);
+    OzoneManagerRatisUtils.setBucketFSOptimized(true);
+
+    OmMetadataManagerImpl metadataManager = new OmMetadataManagerImpl(conf);
+    metadataManager.start(conf);
+
+    org.apache.hadoop.fs.Path effectivePath =
+        new org.apache.hadoop.fs.Path("/");
+
+    Path p = Paths.get(filePath);
+
+    // First get the info about the bucket
+    String bucketKey = metadataManager.getBucketKey(volume, bucket);
+    OmBucketInfo info = metadataManager.getBucketTable().get(bucketKey);
+    long lastObjectId = info.getObjectID();
+    WithParentObjectId objectBucketId = new WithParentObjectId();
+    objectBucketId.setObjectID(lastObjectId);
+    dumpInfo(Type.BUCKET, effectivePath, objectBucketId, bucketKey);
+
+    Iterator<Path> pathIterator =  p.iterator();
+    while(pathIterator.hasNext()) {
+      Path elem = pathIterator.next();
+      String dbKey =
+          metadataManager.getOzonePathKey(lastObjectId, elem.toString());
+      OmDirectoryInfo directoryInfo =
+          metadataManager.getDirectoryTable().get(dbKey);
+      effectivePath = getEffectivePath(effectivePath, elem.toString());
+
+      dumpInfo(Type.INTERMEDIATE_DIR, effectivePath, directoryInfo, dbKey);
```

Comment:
Please add a test case if filePath=/a/b/c/file1. Can you get the path from FileTable if the given path is a **File Type**, here the directoryInfo will return null for the leaf node.